### DISCUSSION TITLE
Error while using `typescript-mongodb` with ts-node.

### DISCUSSION THREAD

**Discussion by GiridharKarnik** [dotansimha/graphql-code-generator#5256] [Q&A] [answered]
Following type-definition throws an error while run using ts-node

schema.grapql

``` graphql
type User @entity {
	id: String @id
	name: UserName! @column
	email: String! @column
	mobileNumber: String! @column
	uid: String @column
	fcmToken: String @column
	accountType: String! @column
	signUpDate: String! @column
}
```

error when run with ts-node

```
[ERROR] 01:43:02 Error: Unknown directive "@entity".

 Unknown directive "@id".
 
 Unknown directive "@column".
 
 Unknown directive "@column".
 
 Unknown directive "@column".
 
 Unknown directive "@column".
 
 Unknown directive "@column".
 
 Unknown directive "@column".
 
 Unknown directive "@column".

```

I am requiring the `schema` file in the following way

``` typescript
const server = new ApolloServer({
    typeDefs: ['./schema.graphql'],
    resolvers
});
```

**Comment by ardatan** [comment, marked as answer]
It isn't related to ts-node. It is a runtime error coming from `makeExecutableSchema`(I guess)
You should add directives to `typeDefs`;
See the docs; https://graphql-code-generator.com/docs/plugins/typescript-mongodb#usage-example
In your case;
```ts
import { DIRECTIVES } from '@graphql-codegen/typescript-mongodb';

const server = new ApolloServer({
    typeDefs: [DIRECTIVES, './schema.graphql'],
    resolvers
});
```

**Comment by GiridharKarnik** [comment]
I assume there are no plugins currently available which will create mongoose schema and models but rather objects for the native mongoDb driver..

**Reply by ardatan** [reply]
No there is not unfortunately. But we'd love to support if anyone wants to contribute :)
